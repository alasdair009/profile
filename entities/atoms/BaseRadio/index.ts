export { BaseRadio } from "./BaseRadio";
export type { BaseRadioProps } from "./BaseRadio";
