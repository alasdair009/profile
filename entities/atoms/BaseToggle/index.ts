export { BaseToggle } from "./BaseToggle";
export type { BaseToggleProps } from "./BaseToggle";
