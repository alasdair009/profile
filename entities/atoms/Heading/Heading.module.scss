@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/lineClamp";
@use "../../../app/styles/typography";
@use "../../../app/styles/helpers/rem";
@use "sass:map";

@mixin append-element($element-name) {
  @at-root #{selector-append($element-name, &)} {
    @content;
  }
}

.root {
  border: none !important;
  color: var(--color) !important;
  font-weight: map.get(typography.$fontWeights, bold);
  margin: 0 auto rem.toRem(map.get(dimensions.$sizes, s24)) !important;
  max-width: rem.toRem(dimensions.$globalTextMaxWidth);
  padding-bottom: 0;
  position: relative;
  text-align: var(--align);
  text-wrap: balance;
  width: fit-content;

  &Clamp {
    @include lineClamp.lineClamp;
  }

  &.Shadow {
    text-shadow: 0 0 rem.toRem(map.get(dimensions.$sizes, s4))
      colors.$colorblackEvil;
  }

  &::after {
    background: colors.$colorGreenGrass;
    bottom: 0;
    content: "";
    display: none;
    height: rem.toRem(3);
    left: 0;
    position: absolute;
    transform: rotate(-2deg);
    width: 100%;
  }

  &.h1 {
    max-width: rem.toRem(map.get(dimensions.$sizes, s512));
    padding-bottom: rem.toRem(map.get(dimensions.$sizes, s8));

    &::after {
      display: block;
    }
  }

  &.h2 {
    max-width: rem.toRem(map.get(dimensions.$sizes, s512));

    &::after {
      display: block;
    }
  }

  @each $headerElement, $headerSize in typography.$headerSizes {
    &.#{$headerElement} {
      font-size: rem.toRem($headerSize);
    }
  }
}
