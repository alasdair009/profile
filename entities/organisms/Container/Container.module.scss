@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";

.root {
  display: block;
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalContentMaxWidth);
  position: relative;
  width: 100%;
}
