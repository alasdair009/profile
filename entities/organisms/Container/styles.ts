import styled from "styled-components";
import { globalContentMaxWidth } from "../../design-tokens/dimensions";
import { rem } from "polished";

export const Root = styled.section`
  display: block;
  margin: 0 auto;
  max-width: ${rem(globalContentMaxWidth)};
  position: relative;
  width: 100%;
`;
