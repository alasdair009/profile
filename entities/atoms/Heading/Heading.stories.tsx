import { Heading } from "./Heading";
import { Meta, StoryObj } from "@storybook/react";

const meta: Meta<typeof Heading> = {
  component: Heading,
  argTypes: {
    children: {
      control: {
        type: "text",
      },
    },
  },
};
export default meta;

export const Default: StoryObj<typeof Heading> = {
  args: {
    children: "Heading",
  },
};
