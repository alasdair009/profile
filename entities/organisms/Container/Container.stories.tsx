import { Container } from "./Container";
import { Heading } from "../../atoms/Heading";
import { Meta, StoryObj } from "@storybook/react";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof Container> = {
  component: Container,
  argTypes: {
    children: {
      type: "string",
    },
  },
  args: {
    children: <Heading>Container</Heading>,
  },
};
export default meta;

export const Default: StoryObj<typeof Container> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const containerElement: HTMLDivElement = canvas.getByTestId(Container.name);

    await expect(containerElement).toBeInTheDocument();
  },
};
