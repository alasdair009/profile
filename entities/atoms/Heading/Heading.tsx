import { Root } from "./styles";
import { HTMLAttributes, ReactNode } from "react";
import { colors, HeadingTypes, TextAlignment } from "@/entities";
import { Property } from "csstype";

export type HeadingProps = {
  /**
   * Visual style of the Heading.
   */
  level?: HeadingTypes;
  /**
   * Heading element level.
   */
  as?: HeadingTypes;
  /**
   * Contents of the heading.
   */
  children: ReactNode;
  /**
   * Text alignment of the heading.
   */
  align?: TextAlignment;
  /**
   * Color of the text.
   */
  color?: Property.Color;
  /**
   * Apply a subtle shadow behind the text.
   */
  textShadow?: boolean;
  /**
   * Number of lines the text is restricted to.
   */
  lines?: number;
} & HTMLAttributes<HTMLHeadingElement>;

/**
 * Text displayed in a heading at a customisable level visually and semantically.
 */
export function Heading({
  level = "h1",
  as,
  align = "left",
  color = colors.whiteGhost,
  textShadow = false,
  lines,
  children,
  ...rest
}: HeadingProps) {
  const headingAs = as ? as : level;
  return (
    <Root
      $color={color}
      $level={level}
      as={headingAs}
      $align={align}
      $textShadow={textShadow}
      $lines={lines}
      data-testid={Heading.name}
      {...rest}
    >
      {children}
    </Root>
  );
}
