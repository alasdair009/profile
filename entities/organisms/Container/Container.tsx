import { Root } from "./styles";
import { HTMLAttributes } from "react";

type HeaderProps = {} & HTMLAttributes<HTMLDivElement>;
export function Container({ children, ...rest }: HeaderProps) {
  return (
    <Root data-testid={Container.name} {...rest}>
      {children}
    </Root>
  );
}
