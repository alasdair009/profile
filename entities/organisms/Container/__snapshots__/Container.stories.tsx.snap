// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/Container Default play-test 1`] = `
<section data-testid="Container"
         class="sc-hDcvty gYfxhu"
>
  <h1 data-testid="Heading"
      class="sc-hfvVTD difGzd"
  >
    Container
  </h1>
</section>
`;
